<launch>
  <!-- Arguments -->
  <arg name="lighthouse_contract" default="airalab.lighthouse.2.robonomics.eth" />
  <arg name="xrt_contract" default="xrt.2.robonomics.eth" />
  <arg name="gas_price_gwei" default="0" />
  <arg name="ens_contract" default="" />
  <arg name="keyfile" default="" />
  <arg name="keyfile_password_file" default="" />


  <arg name="web3_http_provider" default="http://localhost:8545" />
  <arg name="ipfs_http_provider" default="http://127.0.0.1:5001" />

  <!-- Robonomics lighthouse -->
  <group ns="lighthouse">

    <!-- Information channel interface -->
    <include file="$(find robonomics_lighthouse)/launch/infochan.launch">
      <arg name="lighthouse_contract" value="$(arg lighthouse_contract)" />
      <arg name="ipfs_http_provider" value="$(arg ipfs_http_provider)" />
    </include>

    <!-- message signer interface -->
    <include ns="infochan" file="$(find ethereum_common)/launch/signer.launch">
      <arg name="keyfile" value="$(arg keyfile)" />
      <arg name="keyfile_password_file" value="$(arg keyfile_password_file)" />
    </include>

    <!-- XRT token interface -->
    <group ns="xrt">
      <include file="$(find ethereum_common)/launch/erc20.launch">
        <arg name="web3_http_provider" value="$(arg web3_http_provider)" />
        <arg name="erc20_token" value="$(arg xrt_contract)" />
        <arg name="ens_contract" value="$(arg ens_contract)"
             if="$(eval len(arg('ens_contract')) > 0)"/>
        <arg name="keyfile" value="$(arg keyfile)" />
        <arg name="keyfile_password_file" value="$(arg keyfile_password_file)" />
      </include>
    </group>

    <!-- Robonomics order matcher -->
    <node pkg="robonomics_lighthouse" type="matcher_node" name="matcher" output="screen" />

    <!-- Robonomics transaction sender -->
    <node pkg="robonomics_lighthouse" type="lighthouse_node" name="lighthouse" output="screen">
      <param name="lighthouse_abi" textfile="$(find robonomics_lighthouse)/abi/lighthouse.json" />
      <param name="liability_abi" textfile="$(find robonomics_lighthouse)/abi/liability.json" />
      <param name="web3_http_provider" value="$(arg web3_http_provider)" />
      <param name="lighthouse_contract" value="$(arg lighthouse_contract)" />
      <param name="gas_price_gwei" value="$(arg gas_price_gwei)" />
      <param name="keyfile" value="$(arg keyfile)" />
      <param name="keyfile_password_file" value="$(arg keyfile_password_file)" />
      <param name="ens_contract" value="$(arg ens_contract)"
             if="$(eval len(arg('ens_contract')) > 0)"/>
    </node>
  </group>
</launch>
